packages/compiler/src/core/helpers/discriminator-utils.ts

```diff
@@ -28,14 +28,7 @@ export function getDiscriminatedUnion(
   program: Program,
   type: Union,
 ): [DiscriminatedUnion | undefined, readonly Diagnostic[]];
-/**
- * @deprecated Discriminated unions created with `@discriminator` are deprecated.
- * Call `getDiscriminatedUnionFromInheritance` for the legacy model version.
- */
-export function getDiscriminatedUnion(
-  type: Model | Union,
-  discriminator: Discriminator,
-): [DiscriminatedUnionLegacy, readonly Diagnostic[]];
+
 export function getDiscriminatedUnion(
   typeOrProgram: Model | Union | Program,
   typeOrDiscriminator: Union | Discriminator,
```

Comment:
You only have one signature left here. If you want to leave it so that we don't break old callers at runtime that's fine, but no one will be able to call anything other than (Program, Union) now, right?

Reply:
oh yeah good point I can simplify this